fem/doftrans.cpp

```diff
@@ -181,8 +142,54 @@ void StatelessVDofTransformation::InvTransformDual(const Array<int> & face_ori,
    }
 }
 
+void TransformPrimal(const DofTransformation *ran_dof_trans,
+                     const DofTransformation *dom_dof_trans,
+                     DenseMatrix &elmat)
+{
+   if (ran_dof_trans && dom_dof_trans)
+   {
+      ran_dof_trans->TransformPrimalCols(elmat);
+      dom_dof_trans->TransformDualRows(elmat);
+   }
+   else if (ran_dof_trans)
+   {
+      ran_dof_trans->TransformPrimalCols(elmat);
+   }
+   else if (dom_dof_trans)
+   {
+      dom_dof_trans->TransformDualRows(elmat);
+   }
+   else
+   {
+      // If both transformations are NULL this function should not be called
```

Comment:
Not that this is your doing, but do you mind putting an assert to catch this at the top rather than this comment?  Sam for the Dual version as well.

Reply:
Is the correct behavior to error in this case? I would actually think that just returning the original matrix as it does now is what you want. Maybe the comment could be clarified:

```
// Do nothing if both transformations are NULL
```

Reply:
Oh I think you are right.  Folks are using nullptrs to represent identity transformations here.

Reply:
Clarified it in [aa8e466](https://github.com/mfem/mfem/pull/3820/commits/aa8e466e42b36fcb93f2ff9df25345351c30e3f0) as well.